src/linker/Linker/CustomAttributeSource.cs

```diff
@@ -30,13 +32,11 @@ public IEnumerable<CustomAttribute> GetCustomAttributes (ICustomAttributeProvide
 					yield return customAttribute;
 			}
 
-			if (_sources.Count > 0) {
-				foreach (var source in _sources) {
-					if (source.HasCustomAttributes (provider)) {
-						foreach (var customAttribute in source.GetCustomAttributes (provider))
-							yield return customAttribute;
-					}
-				}
+			// This piece of code is optimized for when _xmlCustomAttributes doesn't contain provider
+			// That's why is using ContainsKey and the get indexer instead of TryGetValue
+			if (_xmlCustomAttributes.ContainsKey (provider)) {
```

Comment:
Why not to use TryGet?

Reply:
Visual Studio doesn't show TryGet, neither is in https://referencesource.microsoft.com/#mscorlib/system/collections/generic/dictionary.cs,2e5bc6d8c0f21e67
Maybe I'm missing context on how to use TryGet

Reply:
Sorry, I mean TryGetValue

Reply:
The reason is in the comment - it turns out to be faster for negative cases to call `ContainsKey` - and this will be almost always called on negative cases.

Reply:
I'd prefer simpler code over micro-optimization with explicit comment

Reply:
Let's go with `TryGetValue` - if this becomes a perf problem we can change it in the future.